### PR TITLE
Using CodeStarSourceConnection to GitHub

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

File: templates/account.template.yaml [hunk 1]
```diff
@@ -87,11 +87,7 @@ Resources:
               - Resource: '*'
                 Effect: Allow
                 Action:
-                    - codecommit:GetBranch
-                    - codecommit:GetCommit
-                    - codecommit:UploadArchive
-                    - codecommit:GetUploadArchiveStatus
-                    - codecommit:CancelUploadArchive
+                  - codestar-connections:UseConnection
```
Comment (janbj) [inline templates/account.template.yaml:90]: Not using codecommit anymore, but we need to ad some permissions for codestar-connections

File: templates/extra-project.template.yaml [hunk 2]
```diff
@@ -0,0 +1,120 @@
+AWSTemplateFormatVersion: '2010-09-09'
```
Comment (janbj) [inline templates/extra-project.template.yaml:1]: This is a subset of the full-stack.template.yaml used to register extra projects in the infrastructure

File: templates/full-stack.template.yaml [hunk 3]
```diff
@@ -66,7 +66,7 @@ Parameters:
   BuildImageName:
     Description: Docker image for application build
     Type: String
-    Default: aws/codebuild/amazonlinux2-x86_64-standard:2.0
+    Default: aws/codebuild/standard:3.0
```
Comment (janbj) [inline templates/full-stack.template.yaml:69]: We need a build image with openjdk11 (unless we want to migrate to corretto java at some point)

File: templates/full-stack.template.yaml [hunk 4]
```diff
@@ -91,6 +91,17 @@ Parameters:
     Description: Name of role created by ChildAccountRole template in development account
     Type: String
     Default: ChildAccountRole
+  CodeStarSourceConnectionArn:
```
Comment (janbj) [inline templates/full-stack.template.yaml:94]: Parameters needed for CodeStarSourceConnection

File: templates/full-stack.template.yaml [hunk 5]
```diff
@@ -253,28 +263,91 @@ Resources:
...
-    Description: Username for access to the CodeCommit repository
-    Value: !GetAtt 
-      - ProjectSetupStack
-      - Outputs.GitUserName
-  GitPassword:
-    Description: Password for access to the CodeCommit repository
-    Value: !GetAtt 
-      - ProjectSetupStack
-      - Outputs.GitPassword
+  PipelineArtifactBucketParameter:
```
Comment (janbj) [inline templates/full-stack.template.yaml:266]: Creating parameters for custom data required by the dynamic pipeline creator lambda.

File: templates/lambda.template.yaml [hunk 6]
```diff
@@ -398,8 +398,19 @@ Resources:
               - s3:GetObject
             Resource:
               - !Sub arn:aws:s3:::${QSS3BucketName}/${QSS3KeyPrefix}*
+          - Effect: Allow
+            Action:
+              - ssm:GetParameter
```
Comment (janbj) [inline templates/lambda.template.yaml:403]: We must allow dynamic pipeline maker lambda role access to parameters and codestar connections

File: templates/lambda.template.yaml [hunk 7]
```diff
@@ -398,8 +398,19 @@ Resources:
...
+              - !Sub arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/CICD/*
+          - Effect: Allow
+            Action:
+              - codestar-connections:PassConnection
+            Resource:
+              - !Sub arn:aws:codestar-connections:${AWS::Region}:${AWS::AccountId}:connection/*
 
-  DynamicPipelineMakerLambda:
+
+  DynamicPipelineMakerGitHubLambda:
```
Comment (janbj) [inline templates/lambda.template.yaml]: New lambda to process GitHub Action events

File: templates/project.template.yaml [hunk 8]
```diff
@@ -86,67 +86,20 @@ Resources:
...
-        - arn:aws:iam::aws:policy/AWSCodeCommitPowerUser
-
-  CodecommitCredentials:
-    Type: AWS::CloudFormation::CustomResource
+  CustomDataParameter:
+    Type: AWS::SSM::Parameter
     Properties:
-      ServiceToken: !Ref CodecommitCredentialsLambdaArn
-      IAM_USER_NAME: !Ref RepositoryUser
+      Name: !Sub '/CICD/CustomData/${FullRepositoryId}'
```
Comment (janbj) [inline templates/project.template.yaml:95]: Creating a CustomData parameter for the project, used during dynamic pipeline creation

File: templates/account.template.yaml (hunk 1 above)
Comment (axthosarouris) [reply templates/account.template.yaml:90]: Add a comment regarding the connection of Github and codestar?

**Review by axthosarouris — APPROVED** [review]

**Review by svenngjeto — APPROVED** [review]

File: templates/lambda.template.yaml [hunk 9]
```diff
@@ -433,6 +444,132 @@ Resources:
 
           max_workers = 5
 
+          cf_client = boto3.client('cloudformation')
```
Comment (brinxmat) [inline templates/lambda.template.yaml:452]: In general, not a great fan of inline functions, but for now OK I guess.

Comment (brinxmat) [reply templates/lambda.template.yaml:452]: I note that several of the imports are not used in some of the other lambdas

Comment (brinxmat) [reply templates/lambda.template.yaml:452]: See https://gist.github.com/brinxmat/d3b15e198e3db612c611a8a95f682a5a where I attempt to fix the PEP8 warnings and restructure the code a bit

**Review by brinxmat — CHANGES_REQUESTED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**